### PR TITLE
reduce number of list_models and ensure_server arguments

### PR DESCRIPTION
**Pull request by makelinux**
Use one argument http_client instead of four: tls_insecure, tls_client_cert, tls_client_key, tls_client_passwd.

### CODE REVIEW

**Comment by drpaneas** [comment]
Although the changes do make sense, as they simplify the code a lot by throwing away bunch of None configured TLS dependencies, I see that CI is failing with   curl: (7) Failed to connect to 127.0.0.1 port 8000 after 0 ms: Couldn't connect to server

**Comment by makelinux** [comment]
@drpaneas ,
The real failure is "Error: ilab generate command took more than 20 minutes and was cancelled". I don't know why.

**Comment by leseb** [comment]
@makelinux it's an intermittent failure, just amend and push again to retrigger the CI. Thanks!

**Review by leseb — DISMISSED** [review]

**Comment by markstur** [comment]
@makelinux looks good, but could you address the merge conflicts?

**Comment by makelinux** [comment]
@tiran , Would you like to review please?

**Comment by makelinux** [comment]
@leseb , Could you please review?

**Review by leseb — APPROVED** [review]

**Comment by makelinux** [comment]
@markstur  , Could you please review?

**Review by danmcp — APPROVED** [review]

**Comment by leseb** [comment]
mark's comment was on resolving the conflict, this has been addressed, so dismissed the review. Thanks!

**Comment by leseb** [comment]
@makelinux do you mind sending a followup to @tiran comments? Thanks

File: src/instructlab/model/chat.py [hunk 1]
```diff
@@ -158,10 +158,10 @@ def chat(
...
     endpoint_url,
     api_key,  # pylint: disable=unused-argument
-    tls_insecure,
-    tls_client_cert,
-    tls_client_key,
-    tls_client_passwd,
+    tls_insecure,  # pylint: disable=unused-argument
+    tls_client_cert,  # pylint: disable=unused-argument
+    tls_client_key,  # pylint: disable=unused-argument
+    tls_client_passwd,  # pylint: disable=unused-argument
```
[the remark this answers is not in the document]
Comment (makelinux) [reply src/instructlab/model/chat.py:164]: `chat()` utizes tls parameters via `http_client(ctx.params))`. So they are still used, but not as arguments. I don't know to make it nicer. I know how to make this with `argparse`, not with `click'

File: src/instructlab/model/backends/backends.py [hunk 2]
```diff
@@ -218,10 +216,7 @@ def ensure_server(
     logger: logging.Logger,
     backend: str,
     api_base: str,
-    tls_insecure: bool,
-    tls_client_cert: Optional[str] = None,
-    tls_client_key: Optional[str] = None,
-    tls_client_passwd: Optional[str] = None,
+    http_client=None,
```
[the remark this answers is not in the document]
Comment (makelinux) [reply src/instructlab/model/backends/backends.py:219]: updated in #1435

**Pull request merged**